ts/Stock/Indicators/OBV/OBVIndicator.ts

```diff
@@ -0,0 +1,229 @@
+ * The OBV series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.obv
+ *
+ * @augments Highcharts.Series
+ */
+class OBVIndicator extends SMAIndicator {
+    /**
+     * On-Balance Volume (OBV) technical indicator. This series
+     * requires the `linkedTo` option to be set and should be loaded after
+     * the `stock/indicators/indicators.js` file. Through the `volumeSeriesID`
+     * there also should be linked the volume series.
+     *
+     * @sample stock/indicators/obv
+     *         OBV indicator
+     *
+     * @extends      plotOptions.sma
+     * @since        next
+     * @product      highstock
+     * @requires     stock/indicators/indicators
+     * @requires     stock/indicators/obv
+     * @excluding    allAreas, colorAxis, joinBy, keys, navigatorOptions,
+     *               pointInterval, pointIntervalUnit, pointPlacement,
+     *               pointRange, pointStart, showInNavigator, stacking
+     * @optionparent plotOptions.obv
+     */
+    public static defaultOptions: OBVOptions = merge(SMAIndicator.defaultOptions, {
+        marker: {
+            enabled: false
+        },
+        /**
+         * @excluding index, period
+         */
+        params: {
+            index: void 0 as any,
+            period: void 0 as any,
+            /**
+             * The id of another series to use its data as volume data for the
+             * indiator calculation.
+             */
+            volumeSeriesID: 'volume'
+        },
+        tooltip: {
+            valueDecimals: 0
+        }
+    } as OBVOptions);
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public data: Array<OBVPoint> = void 0 as any;
+    public points: Array<OBVPoint> = void 0 as any;
+    public options: OBVOptions = void 0 as any;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    public getValues<TLinkedSeries extends LineSeries>(
+        series: TLinkedSeries,
+        params: OBVParamsOptions
+    ): (IndicatorValuesObject<TLinkedSeries>|undefined) {
+        const volumeSeries = series.chart.get(params.volumeSeriesID as string),
+            xVal: Array<number> = (series.xData as any),
+            yVal: Array<number> | Array<Array<number>> = (series.yData as any),
+            OBV: Array<Array<number>> = [],
+            xData: Array<number> = [],
+            yData: Array<number> = [],
+            hasOHLC = !isNumber(yVal[0]);
+
+        let OBVPoint: Array<number> = [],
+            i: number = 0,
+            previousOBV: number = 0,
+            curentOBV: number = 0,
+            previousClose: number = 0,
+            curentClose: number = 0,
+            volume: Array<number>,
+            closeValues: Array<number>;
+
+        // Checks if volume series exists.
+        if (volumeSeries) {
+            volume = ((volumeSeries as Series).yData as any);
+
+            for (i; i < yVal.length; i++) {
+                // Add first point and get close value.
+                if (i === 0) {
```

Comment:
You could've calculated the first value before the for loop so that this condition isn't checked in every loop iteration, and then just start the loop from i = 1.